.github/workflows/extension-ci.yml

```
@@ -0,0 +1,203 @@
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - uses: pnpm/action-setup@v4
+        with:
+          version: latest
+
+      - name: Generate cache key
+        id: cache-key
+        run: echo "key=${{ runner.os }}-extension-pnpm-${{ hashFiles('apps/extension/pnpm-lock.yaml') }}" >> $GITHUB_OUTPUT
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:1:97: Double quote to prevent globbing and word splitting

(shellcheck)